crates/redstone/src/core/config.rs

```diff
@@ -1,32 +1,300 @@
-    pub block_timestamp: TimestampMillis,
+    block_timestamp: TimestampMillis,
+}
+
+impl Config {
+    /// Verifies all members of the config.
+    ///
+    /// This method checks whether all config members are correct.
+    ///
+    /// # Arguments
+    ///
+    /// * `signer_count_threshold` - The minimum number of signers required validating the data.
+    /// * `signers` - List of identifiers for signers authorized to sign the data.
+    /// * `feed_ids` - Identifiers for the data feeds from which values are aggregated.
+    /// * `block_timestamp` - The current block time in timestamp format, used for verifying data timeliness.
+    ///
+    /// # Returns
+    ///
+    /// * Success `Self` if arguments to the functions are correct
+    ///   or cresponding Err with `redstone::network::Error` otherwise.
+    pub fn try_new(
+        signer_count_threshold: u8,
+        signers: Vec<SignerAddress>,
+        feed_ids: Vec<FeedId>,
+        block_timestamp: TimestampMillis,
+    ) -> Result<Self, Error> {
+        let config = Self {
+            signer_count_threshold,
+            signers,
+            feed_ids,
+            block_timestamp,
+        };
+
+        config.verify_signer_list()?;
+        config.verify_feed_id_list()?;
+
+        Ok(config)
+    }
+
+    #[inline]
+    fn verify_feed_id_list(&self) -> Result<(), Error> {
+        self.verify_feed_id_list_empty()?;
+        check_no_duplicates(&self.feed_ids)
+            .map_or_else(|| Ok(()), |v| Err(Error::ConfigReocuringFeedId(v)))
+    }
+
+    #[inline(always)]
+    fn verify_feed_id_list_empty(&self) -> Result<(), Error> {
+        if self.feed_ids.is_empty() {
+            return Err(Error::ConfigEmptyFeedIds);
+        }
+
+        Ok(())
+    }
+
+    #[inline]
+    fn verify_signer_list(&self) -> Result<(), Error> {
+        self.verify_signer_count_in_threshold()?;
+        self.verify_signer_count_not_exceeded()?;
+        check_no_duplicates(&self.signers)
+            .map_or_else(|| Ok(()), |v| Err(Error::ConfigReocuringSigner(v)))
+    }
+
+    #[inline(always)]
+    fn verify_signer_count_in_threshold(&self) -> Result<(), Error> {
+        if self.signers.len() < self.signer_count_threshold as usize || self.signers.is_empty() {
+            return Err(Error::ConfigInsufficientSignerCount(
+                self.signers.len() as u8,
+                self.signer_count_threshold,
+            ));
+        }
+
+        Ok(())
+    }
+
+    #[inline(always)]
+    fn verify_signer_count_not_exceeded(&self) -> Result<(), Error> {
+        if self.signers.len() > MAX_SIGNER_COUNT {
+            return Err(Error::ConfigExceededSignerCount(
+                self.signers.len(),
+                MAX_SIGNER_COUNT,
+            ));
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::helpers::hex::{hex_to_bytes, make_feed_id, make_feed_ids};
```

Comment:
add "helpers" feature gate to the mod tests. We should add to the ci just 'cargo test' without any features.